float32 base
float32 shoulder
float32 elbow